docs/v3_spec.md

```diff
@@ -234,6 +235,30 @@ your application. Requests to `your-app.example.com/metrics` will go to the port
 be defined under the `ports` configuration structure. It is also possible to use a port number, but named ports are
 strongly recommended.
 
+### annotations
+
+| **Type** | **Required** |
+|----------|--------------|
+| object   | no           |
+
+A map of annotations to add to the ingress.
+
+Each entry in the ingress list that contains a non-empty `annotations` value will cause a separate Ingress object to
+be created during the deployment. All items that have an empty value will have their hosts/paths merged into a single
+Ingress.
+
+Example:
+```yaml
+ingress:
+  - host: your-app.example.com
+    paths:
+      - path: /foo
+  - host: other.example.com
+    paths:
+      - path: /foo
+    annotations:
+      some/annotation: bar
+```
```

Comment:
The design document has some nice examples of how this feature works; do you think it makes sense to add some more of these examples here too?